workers/analyzer/src/test/kotlin/AnalyzerRunnerTest.kt

```
@@ -422,6 +431,130 @@ class AnalyzerRunnerTest : WordSpec({
+            val inputDir = createOrtTempDir().resolve("project").safeMkdirs()
+
+            val packageCurationProviderConfigs = listOf(
+                ProviderPluginConfiguration(
+                    type = "type1",
+                    options = mapOf("path" to "path1"),
+                    secrets = mapOf("secret1" to "ref1")
+                ),
+                ProviderPluginConfiguration(
+                    type = "type2",
+                    options = mapOf("path" to "path2"),
+                    secrets = mapOf("secret2" to "ref2")
+                )
+            )
+
+            val resolvedPackageCurationProviderConfigs = listOf(
+                ProviderPluginConfiguration(
+                    type = "type1",
+                    options = mapOf("path" to "path1"),
+                    secrets = mapOf("secret1" to "value1")
+                ),
+                ProviderPluginConfiguration(
+                    type = "type2",
+                    options = mapOf("path" to "path2"),
+                    secrets = mapOf("secret2" to "value2")
+                )
+            )
+
+            val environmentConfig = ResolvedEnvironmentConfig(
+                environmentVariables = setOf(
+                    EnvironmentVariableDefinition("ENV_VAR", mockk())
+                )
+            )
+
+            val process = mockk<Process> {
+                every { waitFor() } returns 0
+            }
+
+            val processBuilder = mockk<ProcessBuilder> {
+                every { start() } returns process
+                every { command() } returns listOf("some", "command")
+            }
+
+            val config = AnalyzerJobConfiguration(packageCurationProviders = packageCurationProviderConfigs)
+
+            val runner = spyk(AnalyzerRunner(ConfigFactory.empty()))
+
+            coEvery { runner.createProcessBuilder(any(), any(), any(), any()) } returns processBuilder
+
+            runCatching {
```

The `runCatching` should be removed. Otherwise, exceptions and test failures are silently swallowed.

---

The important part of the test is the `coVerify` block below which verifies that the `runForked` function is called with resolved secrets. The `runBlocking` is currently required because `run` throws an exception, but I did not want to add more setup code which is required for the test.

---

This is not really obvious. Could you put the `run` call in a `shouldThrow` block instead and add a corresponding comment? Then there is at least some control about what is happening.